### PR TITLE
Octos- Ari- Hotel

### PR DESCRIPTION
**Pull request by arrriiii**
# Hotel
Congratulations! You're submitting your assignment!

## Comprehension Questions
Question | Answer
:------------- | :-------------
Describe a design decision you had to make when working on this project. What options were you considering? What helped you make your final decision? | I made a room class but wasn't sure if it had enough responsibility to be it's own class or a hash. I struggled with this idea-- 
Describe a concept that you gained more clarity on as you worked on this assignment. | The idea of coupling dependencies became more clear as I wrote the reservation & room class. Moving forward, I would focus more on writing clear (working) code first and then manage dependencies. 
Describe a _nominal test_ that you wrote for this assignment. | Creating a list of 20 rooms would be a nominal test as it's testing that a room can be created and is an instance of the Room class.
Describe an _edge case test_ that you wrote for this assignment. | I did not write an edge cases.
How do you feel you did in writing pseudocode first, then writing the tests and then the code? | I could have wrote more pseudocode before writing actual code. I initially developed an outline of the different classes on paper and that was helpful.

### CODE REVIEW

**Comment by kariabancroft** [comment]
@Ari-1 You need to answer the comprehension questions - then we can review the code!

File: lib/reservation.rb [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+      if @end_date < @start_date
+        raise ArgumentError('Invalid Date Range')
+      end
+
+      @reservations << input
+    end
+
+    def self.all
+      return @reservations
```
Comment (kariabancroft) [inline lib/reservation.rb]: You cannot use instance variables in class methods because they refer to the overall class.

File: lib/front_desk.rb [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    # to create a reservation
+    def add_room(start_date, end_date)
+      if valid_date?(start_date)
+        if valid_date?(end_date)
+          new_ticket = {}
+          new_ticket[:ticket_id] = reservation_data.length + 1
+          new_ticket[:start_date] = Date.parse(start_date)
+          new_ticket[:end_date] = Date.parse(end_date)
+
+          @total_rooms.find do |room|
```
Comment (kariabancroft) [inline lib/front_desk.rb]: Why use `find` here? Could you use `room.sample` by itself without the `find` method?

**Comment by kariabancroft** [comment]
# Hotel
## What We're Looking For

Feature | Feedback
---     | ---
**Design** |
Demonstrated classes having a single responsibility | You structured your classes such that building out the functionality should keep them separated out.
Demonstrated loose coupling |Yes
Methods demonstrate a good use of encapsulation, inputs and outputs | Yes
Wave 1 requirements |Yes
Wave 2 requirements |No
Wave 3 requirements |No

For a project of this length, you should be committing using git much more often. For the amount of code you had written, you had a very insufficient number of tests.
